.github/workflows/ci.yml

```
@@ -2,34 +2,228 @@ name: CI/CD Pipeline
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install uv
+      uses: astral-sh/setup-uv@v4
+      with:
+        version: "latest"
 
-    - name: Set up Python
-      uses: actions/setup-python@v2
+    - name: Cache uv dependencies
+      uses: actions/cache@v3
```

actionlint (1.7.7)

the runner of "actions/cache@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)